lib/ansible/modules/monitoring/zabbix_map.py

```diff
@@ -0,0 +1,800 @@
+        elif "label" in data:
+            label = remove_quotes(data['label'])
+        else:
+            label = default
+        return label
+
+    def _get_map_id(self, map_name):
+        exist_map = self._zapi.map.get({'filter': {'name': map_name}})
+        if exist_map:
+            return exist_map[0]['sysmapid']
+        return None
+
+    def _get_group_id(self, group_name):
+        exist_group = self._zapi.hostgroup.get({'filter': {'name': group_name}})
+        if exist_group:
+            return exist_group[0]['groupid']
+        return None
+
+    def map_exists(self):
+        return bool(self.map_id)
+
+    def create_map(self, map_config):
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            result = self._zapi.map.create(map_config)
+            if result:
+                return result
+        except Exception as e:
+            self._module.fail_json(msg="Failed to create map: %s" % e)
+
+    def update_map(self, map_config):
+        if not self.map_id:
+            self._module.fail_json(msg="Failed to update map: map_id is unknown. Try to create_map instead.")
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            map_config['sysmapid'] = self.map_id
+            result = self._zapi.map.update(map_config)
+            if result:
+                return result
+        except Exception as e:
+            self._module.fail_json(msg="Failed to update map: %s" % e)
+
+    def delete_map(self):
+        if not self.map_id:
+            self._module.fail_json(msg="Failed to delete map: map_id is unknown.")
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            self._zapi.map.delete([self.map_id])
+        except Exception as e:
+            self._module.fail_json(msg="Failed to delete map, Exception: %s" % e)
+
+    def is_exist_map_correct(self, generated_map_config):
+        exist_map_configs = self._zapi.map.get({
+            'sysmapids': self.map_id,
+            'selectLinks': 'extend',
+            'selectSelements': 'extend'
+        })
+        exist_map_config = exist_map_configs[0]
+        if not self._is_dicts_equal(generated_map_config, exist_map_config):
+            return False
+        if not self._is_selements_equal(generated_map_config['selements'], exist_map_config['selements']):
+            return False
+        self._update_ids(generated_map_config, exist_map_config)
+        if not self._is_links_equal(generated_map_config['links'], exist_map_config['links']):
+            return False
+        return True
+
+    def _is_selements_equal(self, generated_selements, exist_selements):
+        if len(generated_selements) != len(exist_selements):
+            return False
+        generated_selements_sorted = sorted(generated_selements, key=itemgetter('elementid', 'label'))
```

Comment:
elementid no longer exists in the Zabbix 3.4 API on the map object.
Would selementid suffice here?

(this crashes when used with Zabbix 3.4)